graql/reasoner/query/CompositeQuery.java

```diff
@@ -311,9 +343,26 @@ public void checkValid() {
 
     @Override
     public Conjunction<Pattern> getPattern() {
-        Set<Pattern> pattern = Sets.newHashSet(getConjunctiveQuery().getPattern());
-        getComplementQueries().stream().map(ResolvableQuery::getPattern).forEach(pattern::add);
-        return Graql.and(pattern);
+        if (pattern == null) {
+            Set<Pattern> conjunctPatterns = Sets.newLinkedHashSet(getConjunctiveQuery().getPattern().getPatterns());
+            getComplementQueries().stream().map(ResolvableQuery::getPattern).forEach(p -> {
+                if (p instanceof Conjunction && ((Conjunction<Pattern>) p).getPatterns().size() == 1) {
```

Comment:
Investigate whether this can be lived with by adding extra `{}` inside the `not` in the test case

Reply:
This was achievable!